starter-code/models/user.model.js

```diff
@@ -0,0 +1,42 @@
+const mongoose = require('mongoose');
+const bcrypt = require('bcrypt');
+const SALT_WORK_FACTOR = 10;
+
+
+const userSchema = new mongoose.Schema({
+    username: {
+        type: String,
+        require: [true, "Username is required"],
+        unique: true
+    },
+    password: {
+        type: String,
+        required: [true, "User needs a password"]
+    }
+}, { timestamps: true });
+
+// userSchema.pre('save', (next) => { }) NOFUNCIONA
```

Comment:
la particularidad de las arrow functions son que no pierdes el contexto. Por esto no te funciona, dentro de la función estás intentando asignar a la constante user el this, y ese this con arrow function es un contexto que es undefined, porque no tiene nadie por encima. Por eso se siguen usando las funciones normales para cuando quieres que tenga contexto diferente